packages/settingeditor/src/pluginlist.tsx

```diff
@@ -249,67 +228,140 @@ namespace Private {
     return typeof hint === 'string' ? hint : '';
   }
 
+  setFilter(filter: (item: string) => boolean) {
+    this._filter = (value: ISettingRegistry.IPlugin) => {
+      return filter(value.schema.title?.toLowerCase() ?? '');
+    };
+    this.update();
+  }
+
+  setError(id: string, error: boolean) {
+    if (this._errors[id] !== error) {
+      this._errors[id] = error;
+      this.update();
+    } else {
+      this._errors[id] = error;
+    }
+  }
+
+  mapPlugins(plugin: ISettingRegistry.IPlugin) {
+    const { id, schema, version } = plugin;
+    const trans = this.translator.load('jupyterlab');
+    const title =
+      typeof schema.title === 'string' ? trans._p('schema', schema.title) : id;
+    const description =
+      typeof schema.description === 'string'
+        ? trans._p('schema', schema.description)
+        : '';
+    const itemTitle = `${description}\n${id}\n${version}`;
+    const icon = this.getHint(ICON_KEY, this.registry, plugin);
+    const iconClass = this.getHint(ICON_CLASS_KEY, this.registry, plugin);
+    const iconTitle = this.getHint(ICON_LABEL_KEY, this.registry, plugin);
+
+    return (
+      <button
+        onClick={this._evtMousedown}
+        className={`${
+          id === this.selection
+            ? 'jp-mod-selected jp-PluginList-entry'
+            : 'jp-PluginList-entry'
+        } ${this._errors[id] ? 'jp-ErrorPlugin' : ''}`}
+        data-id={id}
+        key={id}
+        title={itemTitle}
+      >
+        {id === this.selection || this._errors[id] ? (
+          <div className="jp-SelectedIndicator" />
+        ) : undefined}
+        <LabIcon.resolveReact
+          icon={icon || (iconClass ? undefined : settingsIcon)}
```

Comment:
```suggestion
          icon={icon || (iconClass ? null : settingsIcon)}
```

Reply:
This doesn't compile because of a type error

Reply:
```
Type 'string | LabIcon | null' is not assignable to type 'IMaybeResolvable'.
  Type 'null' is not assignable to type 'IMaybeResolvable'.ts(2322)
```

Reply:
Ok then let be it `undefined`